app/views/common/BackboneModalHarness.vue

```diff
@@ -53,6 +53,7 @@
           ...this.$props.modalArgs,
```

Comment:
Tangental thoughts regarding modals in Vue. We are currently adding modals to the TeacherDashboard. Interested in your thoughts regarding this sort of dependency injection to show modals in a vue only environment? I like being able to instantiate a modal by importing it and calling a method.

Currently we register the modals in a parent component, and then show them based on booleans which is not scalable. I need to do some more digging for solutions that don't require v-show or v-if checks on the modal. Or maybe that is a fine way to go.

Reply:
Happy to talk more about this IRL.  I added this harness to allow us to bind the modals built with Backbone / CocoView to be able to conform to the Vue component lifecycle so that we can ensure we're opening / showing / closing / hiding the modals as appropriate with the different Vue lifecycle methods.  It also has the side effect of being able to open / close modals in a Vue way.  No other reasons and I'm not fully understanding the different options you're looking at but happy to discuss more!